var/spack/repos/builtin/packages/intel-oneapi-mkl/package.py

```diff
@@ -111,6 +111,8 @@ class IntelOneapiMkl(IntelOneApiLibraryPackage):
         expand=False,
     )
 
+    variant("gfortran", default=False, description="Compatibility with Gnu Fortran")
```

Comment:
I think you can:
```suggestion
    depends_on("fortran", type="build")
```
here...

Reply:
What would happen if they did not have a fortran compiler configured?

Reply:
If that needs to be a variant, you can:
```python
variant("fortran", default=..., description="Enable Fortran extension to the package")
depends_on("fortran", type="build", when="+fortran")
```

Reply:
`hdf5` is an example for that.

Reply:
After merge, will follow up with @alalazo on more precise modeling of oneapi runtime constraints.